ccache install broken since upgrade to 4.10.2

Since https://github.com/sagemath/sage/pull/39119, ccache cannot be installed because it tries to download xxhash at install time. @gmou3 @kwankyu @vbraun 

https://github.com/passagemath/passagemath/actions/runs/12532079587/job/34950722473#step:11:2642
```
  [ccache-4.10.2]   [spkg-install] -- Using src='https://github.com/Cyan4973/xxhash/archive/refs/tags/v0.8.2.tar.gz'
  [ccache-4.10.2]   [spkg-install] -- Retrying...
  [ccache-4.10.2]   [spkg-install] -- Using src='https://github.com/Cyan4973/xxhash/archive/refs/tags/v0.8.2.tar.gz'
  [ccache-4.10.2]   [spkg-install] -- Retry after 5 seconds (attempt #2) ...
  [ccache-4.10.2]   [spkg-install] -- Using src='https://github.com/Cyan4973/xxhash/archive/refs/tags/v0.8.2.tar.gz'
  [ccache-4.10.2]   [spkg-install] -- Retry after 5 seconds (attempt #3) ...
  [ccache-4.10.2]   [spkg-install] -- Using src='https://github.com/Cyan4973/xxhash/archive/refs/tags/v0.8.2.tar.gz'
  [ccache-4.10.2]   [spkg-install] -- Retry after 15 seconds (attempt #4) ...
  [ccache-4.10.2]   [spkg-install] -- Using src='https://github.com/Cyan4973/xxhash/archive/refs/tags/v0.8.2.tar.gz'
  [ccache-4.10.2]   [spkg-install] -- Retry after 60 seconds (attempt #5) ...
  [ccache-4.10.2]   [spkg-install] -- Using src='https://github.com/Cyan4973/xxhash/archive/refs/tags/v0.8.2.tar.gz'
  [ccache-4.10.2]   [spkg-install] CMake Error at xxhash-subbuild/xxhash-populate-prefix/src/xxhash-populate-stamp/download-xxhash-populate.cmake:163 (message):
  [ccache-4.10.2]   [spkg-install]   Each download failed!
  [ccache-4.10.2]   [spkg-install] 
  [ccache-4.10.2]   [spkg-install]     error: downloading 'https://github.com/Cyan4973/xxhash/archive/refs/tags/v0.8.2.tar.gz' failed
  [ccache-4.10.2]   [spkg-install]           status_code: 28
  [ccache-4.10.2]   [spkg-install]           status_string: "Timeout was reached"
  [ccache-4.10.2]   [spkg-install]           log:
  [ccache-4.10.2]   [spkg-install]           --- LOG BEGIN ---
  [ccache-4.10.2]   [spkg-install]           Uses proxy env variable https_proxy == 'http://192.0.2.0:5187/'
  [ccache-4.10.2]   [spkg-install]     Trying 192.0.2.0:5187...
```

--- a/build/pkgs/ccache/spkg-install.in
+++ b/build/pkgs/ccache/spkg-install.in
@@ -1,5 +1,8 @@
 cd src
 
+# Allow downloading dependencies
+unset https_proxy
+
 mkdir build
 cd build
 sdh_cmake -DCMAKE_BUILD_TYPE=Release \

